### ISSUE TITLE
Run tests in parallel and calculate coverage

### ISSUE THREAD

**Issue by ajp-io** [onsi/ginkgo#393]
I don't seem to be able to use the `-p` and `-cover` options in tandem. When I do, the coverage is not calculated. I assume there might be something that prevents the coverage from being calculated in parallel, but I want to ask anyway. Thanks.

**Comment by williammartin** [comment]
Hey, sorry it took so long to respond. This is interesting but can you be more specific about "the coverage is not calculated". Seems like we have a test that is supposed to cover this (https://github.com/onsi/ginkgo/blob/master/integration/coverage_test.go#L88-L100) but I can see it doesn't do anything more than check a file exists.

**Comment by alexparker215** [comment]
I mean that normally when you use `-cover` it prints out "coverage: 95.0% of statements". But if I use the `-p` option in tandem with `-cover`, then it doesn't print out the coverage.

**Comment by fjorgemota** [comment]
What's "waiting for response" on this issue?

I miss this feature too. Just to mention an example, I need to run tests sequentially if I want to get coverage % printed directly in the end of the `ginkgo` command, meanwhile, if I run all tests in parallel using `ginkgo -p -cover`, only the coverage file is generated, but without printing the total % in the screen.

I searched on Ginkgo's code and seems like this message is not printed directly by Ginkgo...Is that true? If that's the case, I think we will need to get the coverage calculated by each parallel process started, and then calculate how much coverage we have to print on the screen. Is that the case?

**Comment by sazary** [comment]
I'm experiencing the same issue. seems like `-p` and `-cover` are incompatible 

```
➜  ginkgo -cover
Running Suite: Accept Suite
===========================
Random Seed: 1561189697
Will run 4 of 4 specs

••••
Ran 4 of 4 Specs in 0.001 seconds
SUCCESS! -- 4 Passed | 0 Failed | 0 Pending | 0 Skipped PASS
coverage: 51.7% of statements

Ginkgo ran 1 suite in 2.563600425s
Test Suite Passed



➜  ginkgo -p
Running Suite: Accept Suite
===========================
Random Seed: 1561189708
Will run 4 specs

Running in parallel across 4 nodes

••••
Ran 4 of 4 Specs in 0.018 seconds
SUCCESS! -- 4 Passed | 0 Failed | 0 Pending | 0 Skipped


Ginkgo ran 1 suite in 2.895559714s
Test Suite Passed



➜ ginkgo -cover -p
Running Suite: Accept Suite
===========================
Random Seed: 1561189715
Will run 4 specs

Running in parallel across 4 nodes

••••
Ran 4 of 4 Specs in 0.011 seconds
SUCCESS! -- 4 Passed | 0 Failed | 0 Pending | 0 Skipped


Ginkgo ran 1 suite in 1.982561045s
Test Suite Passed
```

**Comment by onsi** [comment]
I'm working through the backlog of old Ginkgo issues - apologies as this issue is probably stale now.

I'm improving how Ginkgo supports profiles, and coverage analysis, in [v2](#711).  I'm planning to fix this with the 2.0 release.

I'm going to close this for now, but feel free to reopen it if you'd like.

**Comment by onsi** [comment]
this is now fixed on the v2 branch
